src/basilisp/lang/interfaces.py

```diff
@@ -374,6 +374,18 @@ def pop(self: Self) -> Self:
         raise NotImplementedError()
 
 
+class IReduce:
+    @abstractmethod
+    def reduce(self, f, init):
+        raise NotImplementedError()
+
+
+class IReduceKV:
+    @abstractmethod
+    def reduce_kv(self, f, init):
+        raise NotImplementedError()
```

Comment:
Even though they are not used in the Basilisp code, we fully type hint every interface. These should be generic. Please review other interfaces for examples.

Reply:
Done. I'm not entirely sure I did it right, the functions are pretty open ended.